core/src/main/scala/org/apache/spark/sql/execution/columnar/ColumnBatch.scala

```diff
@@ -85,30 +85,40 @@ abstract class ResultSetIterator[A](conn: Connection,
   def close() {
     // if (!hasNextValue) return
     try {
-      if (rs ne null) {
-        // GfxdConnectionWrapper.restoreContextStack(stmt, rs)
-        // rs.lightWeightClose()
-        rs.close()
+      try {
+        if (rs ne null) {
+          // GfxdConnectionWrapper.restoreContextStack(stmt, rs)
+          // rs.lightWeightClose()
+          rs.close()
+        }
+      } catch {
+        case NonFatal(e) => logWarning("Exception closing resultSet", e)
       }
-    } catch {
-      case NonFatal(e) => logWarning("Exception closing resultSet", e)
-    }
-    try {
-      if (stmt ne null) {
-        stmt.getConnection match {
-          case embedConn: EmbedConnection =>
-            val lcc = embedConn.getLanguageConnection
-            if (lcc ne null) {
-              lcc.clearExecuteLocally()
-            }
-          case _ =>
+      try {
+        if (stmt ne null) {
+          stmt.getConnection match {
+            case embedConn: EmbedConnection =>
+              val lcc = embedConn.getLanguageConnection
+              if (lcc ne null) {
+                lcc.clearExecuteLocally()
+              }
+            case _ =>
+          }
+          stmt.close()
+        }
+      } catch {
+        case NonFatal(e) => logWarning("Exception closing statement", e)
+      }
+      hasNextValue = false
+    } finally {
+      try {
+        if (conn != null) {
+          conn.close()
```

Comment:
Looked at the calls to ResultSetIterator. Two of the paths probably require the connection close but the one from ColumnBatchIteratorOnRS is closed separately in SmartConnectorColumnRDD AFTER the commit is called, so you cannot close the connection here. So you probably need a boolean in the constructor whether to close the connection or not and pass as false at least from ColumnBatchIteratorOnRS.

@suranjan Please check this if there was some deliberate reason why connection close was removed for all paths. From what I recall this code used to have a connection close earlier.